packages/nest-commander/src/completion.factory.ts

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { addCompletionSpecCommand } from '@fig/complete-commander';
+import { INestApplicationContext } from '@nestjs/common';
+import { Command } from 'commander';
+import { CompletionFactoryOptions } from './completion.factory.interface';
+import {
+  Commander,
+  COMPLETION_SH_TEMPLATE,
+  COMPLETION_ZSH_TEMPLATE,
+} from './constants';
+
+/**
+ * @description Factory to generate completion script for BASH and ZSH
+ */
+export class CompletionFactory {
```

Comment:
Out of curiosity, would there be a way to have all of this hidden behind a flag in the `CommandFactory`'s options? So that a dev could do `await CommandFactory.run(RootModule, { completions: true })` and we'd call this `CompletionFactory` before getting to the running of the command?

Reply:
That would be quite a challenge, because the completion configuration requires more inputs rather than just a Boolean

Reply:
We could do something like 
```ts
await CommandFactory.run(
  RootModule,
  {
    completions: {
      cmd: 'CLI-NAME',
      fig: false,
    }
  }
)
```
So that if the `completions` option exists we just pass it directly to the `CompletionFactory`. Really just looking at the DX here, as it would be nice to have a simple `main.ts` for the CLI whenever possible. I know the current approach works though, so if it is too much we might just make that a possible future improvement

Reply:
Yeah, I see your point. 
We could try something like that, but it will require more inputs. 

I'll push a change so we can review it and then decide 🤟